--- a/src/ocat.c	2018-05-24 08:33:11.479301567 +0300
+++ b/src/ocat.c	2018-05-24 08:53:57.034571026 +0300
@@ -38,6 +38,7 @@
          "   -I                    GarliCat mode, use I2P instead of Tor\n"
          "   -l [<ip>:]<port>      set ocat listen address and port, default = 127.0.0.1:%d\n"
          "   -L <log_file>         log output to <log_file> (default = stderr)\n"
+         "   -n <tunname>          set the tun device name, may contain format string (e.g. tun%%d)\n"
          "   -o <ipv6_addr>        convert IPv6 address to onion url and exit\n"
          "   -p                    use TAP device instead of TUN\n"
          "   -P [<pid_file>]       create pid file at location of <pid_file> (default = %s)\n"
@@ -335,7 +336,7 @@
    log_debug("parse_opt_early()");
    opterr = 1;
    optind = 1;
-   while ((c = getopt(argc, argv, "f:IabBCd:e:hHrRiopl:t:T:s:Uu:4L:P:")) != -1)
+   while ((c = getopt(argc, argv, "f:IabBCd:e:hHrRiopl:t:T:s:Uu:4L:P:n:")) != -1)
    {
       log_debug("getopt(): c = %c, optind = %d, opterr = %d, optarg = \"%s\"", c, optind, opterr, SSTR(optarg));
       switch (c)
@@ -437,6 +438,10 @@
          case 'T':
             tun_dev_ = optarg;
             break;
+
+         case 'n':
+            strncpy(CNF(tunname), optarg, sizeof(CNF(tunname)));
+            break;
 #endif
 
          case 'U':
