diff --git a/DaemonLinux.cpp b/DaemonLinux.cpp
index 22d7dec..ed667fd 100644
--- a/DaemonLinux.cpp
+++ b/DaemonLinux.cpp
@@ -76,9 +76,9 @@ namespace i2p
 				}
 
 				// point std{in,out,err} descriptors to /dev/null
-                stdin  = freopen("/dev/null", "r", stdin);
-				stdout = freopen("/dev/null", "w", stdout);
-				stderr = freopen("/dev/null", "w", stderr);
+				freopen("/dev/null", "r", stdin);
+				freopen("/dev/null", "w", stdout);
+				freopen("/dev/null", "w", stderr);
 			}
 
 			// Pidfile
